### PR TITLE
Fix/better error logging

### PR DESCRIPTION
**Pull request by alfonso-presa**
## Proposed changes

Log status code on errors running commands (or starting the session). It makes easier to debug problems with not found (404), or access error (401) when configuration is wrong.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

- [X] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

### CODE REVIEW

File: packages/webdriver/src/request.js [hunk 1]
```diff
@@ -154,7 +154,7 @@ export default class WebDriverRequest extends EventEmitter {
          * retry, e.g. if sessionId is invalid
          */
         if (retryCount >= totalRetryCount || error.message.includes('invalid session id')) {
-            log.error('Request failed due to', error)
+            log.error('Request failed with status', response.statusCode, 'due to', error)
```
Comment (christian-bromann) [inline packages/webdriver/src/request.js:157]: How about:
```suggestion
            log.error(`Request failed with status ${response.statusCode} due to ${error.stack}`)
```

File: packages/webdriver/src/request.js [hunk 2]
```diff
@@ -163,7 +163,7 @@ export default class WebDriverRequest extends EventEmitter {
 
         ++retryCount
         this.emit('retry', { error, retryCount })
-        log.warn('Request failed due to', error.message)
+        log.warn('Request failed with status', response.statusCode, 'due to', error.message)
```
Comment (christian-bromann) [inline packages/webdriver/src/request.js:166]: same:
```suggestion
        log.warn(`Request failed with status ${response.statusCode} due to ${error.message}`)
```

File: packages/webdriver/src/request.js (hunk 1 above)
Comment (alfonso-presa) [reply packages/webdriver/src/request.js]: Done :-)

File: packages/webdriver/src/request.js (hunk 2 above)
Comment (alfonso-presa) [reply packages/webdriver/src/request.js]: Done too :-)

**Pull request merged**